### PR TITLE
New plugin creater

### PR DESCRIPTION
**Pull request by SimonBaeumer**
Supporting the `5.2` and `legacy` plugin system for `sw plugin:create`.
Working on top of @mohrekopp pull request: #70. 

Added:
- Move `Views/` to `Resources/views`
- Move `Snippets/` to `Resources/snippets`
- `Resources/menu.xml`
- `Resources/config.xml`
- Service registrations in `Resources/services.xml`
- Create widget
- Separate directories for `legacy` and `5.2` plugin system
- Little unit test fixes

Thanks @mohrekopp for your Pull-Request #70, these big changes must be done due to missing documentation at the moment.

### CODE REVIEW

**Comment by FiveDigital** [comment]
Shouldn't Commands and Subscribers be written in the `services.xml` with their according tags instead of adding methods in the plugin class for them?

File: src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        ];
+<?php } ?>
+    }
+<?php if ($configuration->hasCommands) { ?>
+    /**
+     * Register PluginCommands
+     *
+     * @param Application $application
+     */
+    public function registerCommands(Application $application)
```
Comment (mitelg) [inline src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl]: is this necessary?

**Comment by mitelg** [comment]
@FiveDigital commands and subscribers are registered in the services.xml if I see this correctly. the onStartDispatch event in the plugin.php is only used for registering the API components

**Comment by FiveDigital** [comment]
The according part was removed exactly one minute after my comment with https://github.com/shopwareLabs/sw-cli-tools/pull/71/files/1f5edd94f005e0ff43386f9bd8ea204fe851216b..9b4004c391cafc68622126a0ccb114d7dc4da4e0 👍

File: src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl (hunk 1 above)
Comment (mohrekopp) [inline src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl]: IMHO we should actually remove this part

Comment (SimonBaeumer) [inline src/Extensions/Shopware/PluginCreator/template/current/PluginClass.tpl]: Agreed 100% ;)

**Comment by SimonBaeumer** [comment]
I improved the pull-request a little bit more by adding the `Subscriber/ApiSubscriber.php` and better support for the command arguments.

**Comment by OliverSkroblin** [comment]
1. new Plugins created in root directory

2. install and uninstall function in plugin class are empty, unnecessary boiler plate code (php storm override function generates this code within 2sec )

3. controller path service definition exists even if when the controllerPath subscriber class not exist.

4. --have-filter 
	4.0 please consider SearchBundleES too
	4.1 Add a decorator for ESIndexingBundle/Product/*
	4.2 Add a sorting too please, with template extension  
	4.3 move CriteriaRequestHandler into SearchBundle namespace
5. --have-widget
	5.0 Mixed with backend module implementation
	5.2 ControllerPath Subscriber contains template extends command for backend widgets => extract
	5.3 Backend/Application controller contains php code for backend widget
	5.4 views/backend/application/widgets contains extjs widgets 
	==> please split in seperate namespace
6. $this->container->get('application')->Loader() => $this->container->get('Loader');

**Comment by SimonBaeumer** [comment]
@OliverSkroblin The `1.` and `2.` point of your list should be done in additional PRs, because this PR just handles  the compatibility to the `5.2` plugin system. 
In this case the target installation directory and SearchBundleES integration would be new functionality.

All other points on your list were fixed.

**Comment by SimonBaeumer** [comment]
Added:
 
 - Root folder detection, created plugins will be created in the plugin path
  - legacy plugins in `engine/Shopware/Plugins/Local/your_namespace' 
  - `5.2` plugins in `custom/plugins`
  - if the command will not be executed from shopware root it will be created in the same directory
- Added elastic search integration
- Added product search decorator
- Added sorting

**Comment by SimonBaeumer** [comment]
This PR is ready to merge. 
Thank you for your contribution, @mohrekopp ;)

**Pull request merged**